### PR TITLE
Docs: Fundamentals of Block Development - File structure of a block

### PR DESCRIPTION
**Pull request by juanmaguitar**
> [!NOTE]  
> I opened this new PR to avoid some unwanted commits I got on the original one https://github.com/WordPress/gutenberg/pull/56316 when trying to get that branch rebased from trunk

Add documentation for file structure of a block
This PR closes the issue https://github.com/WordPress/gutenberg/issues/55437

### CODE REVIEW

**Comment by juanmaguitar** [comment]
@fabiankaegy as you approved https://github.com/WordPress/gutenberg/pull/56316, can you have a look at this PR and approve it if everything remains OK?
Pinging also @carolinan as you were part of the review of https://github.com/WordPress/gutenberg/pull/56316

I'll merge it once I get the proper approvals without adding it to the sidebar.
I'll add the pages available for  the "Fundamentals of Block Development" to the sidebar (add them to the manifesto) with a PR for https://github.com/WordPress/gutenberg/issues/56282 once a few of the pages described on https://github.com/WordPress/gutenberg/issues/56282 have been merged

File: docs/getting-started/fundamentals-block-development/file-structure-of-a-block.md
Comment (ryanwelcher) [inline docs/getting-started/fundamentals-block-development/file-structure-of-a-block.md]: I believe 
```suggestion
[`package.json`](https://docs.npmjs.com/cli/v10/configuring-npm/package-json) is a configuration file for a Node.js project. In this file you define the NPM dependencies of the block and the scripts used for local work.
```

I believe it's supposed to be node.js

File: docs/getting-started/fundamentals-block-development/file-structure-of-a-block.md
Comment (ryanwelcher) [inline docs/getting-started/fundamentals-block-development/file-structure-of-a-block.md:50]: `save` is a standard JavaScript function. It can be created as a component but under the hood only the render method is called as a JS function. Looking at scaffolded files, you'll notice that `save` is lowercase and React components are written in uppercase.

This may be pedantic but it's worth mentioning.

**Review by ryanwelcher — APPROVED** [review]
I left a couple of notes that are probably not critical so feel free to merge if you disagree.

File: docs/getting-started/fundamentals-block-development/file-structure-of-a-block.md
Comment (ndiego) [inline docs/getting-started/fundamentals-block-development/file-structure-of-a-block.md]: This line feels out of place. I am not sure how it directly relates to the file structure.  🤔

Comment (juanmaguitar) [reply docs/getting-started/fundamentals-block-development/file-structure-of-a-block.md]: You're right. It's not clear the value of this message here.
I'll remove it.

**Pull request merged**
